raiden_contracts/tests/test_service_registry.py

```diff
@@ -2,49 +2,80 @@
 
 import pytest
 from eth_tester.exceptions import TransactionFailed
-from web3.contract import Contract
+from web3 import Web3
+from web3.contract import Contract, get_event_data
 
-from raiden_contracts.tests.utils.constants import CONTRACT_DEPLOYER_ADDRESS
+from raiden_contracts.constants import CONTRACT_SERVICE_REGISTRY, EVENT_REGISTERED_SERVICE
+from raiden_contracts.contract_manager import ContractManager, contracts_precompiled_path
+from raiden_contracts.tests.utils.constants import EMPTY_ADDRESS, SERVICE_DEPOSIT
 
 
-def test_owner_of_service_registry(service_registry: Contract) -> None:
-    """ The owner of ServiceRegistry should be the deployer """
-    assert service_registry.functions.owner().call() == CONTRACT_DEPLOYER_ADDRESS
+def test_deposit_contract(
+    get_deposit_contract: Callable, custom_token: Contract, get_accounts: Callable
+) -> None:
+    (A,) = get_accounts(1)
+    custom_token.functions.mint(100).call_and_transact({"from": A})
+    depo = get_deposit_contract([custom_token.address, 0, A])
+    custom_token.functions.transfer(depo.address, 100).call_and_transact({"from": A})
+    assert custom_token.functions.balanceOf(A).call() == 0
+    assert custom_token.functions.balanceOf(depo.address).call() == 100
+    depo.functions.withdraw(A).call_and_transact({"from": A})
+    assert custom_token.functions.balanceOf(A).call() == 100
+    assert custom_token.functions.balanceOf(depo.address).call() == 0
 
 
 def test_deposit(
     service_registry: Contract, custom_token: Contract, get_accounts: Callable
 ) -> None:
     (A,) = get_accounts(1)
-    custom_token.functions.mint(10).call_and_transact({"from": A})
-    custom_token.functions.approve(service_registry.address, 10).call_and_transact({"from": A})
+    custom_token.functions.mint(SERVICE_DEPOSIT).call_and_transact({"from": A})
+    custom_token.functions.approve(service_registry.address, SERVICE_DEPOSIT).call_and_transact(
+        {"from": A}
+    )
 
     # happy path
-    service_registry.functions.deposit(10).call_and_transact({"from": A})
-    assert service_registry.functions.deposits(A).call() == 10
-    assert custom_token.functions.balanceOf(A).call() == 0
+    old_balance = custom_token.functions.balanceOf(A).call()
```

Comment:
There's a second happy path when extending the validity of an existing slot. At least a rough test for that case would make sense.

Reply:
I agree.  I'll do that as a continuation of this case.